src/com/dotmarketing/portlets/htmlpageasset/business/HTMLPageAssetAPIImpl.java

```diff
@@ -473,57 +447,107 @@ public boolean migrateAllLegacyPages(final User user, boolean respectFrontEndPer
     			}
     	return result;
     }
-    
+
     @Override
     public HTMLPageAsset migrateLegacyPage(HTMLPage legacyPage, User user, boolean respectFrontEndPermissions) throws Exception {
         Identifier legacyident=APILocator.getIdentifierAPI().find(legacyPage);
         VersionInfo vInfo=APILocator.getVersionableAPI().getVersionInfo(legacyident.getId());
-        
-        List<HTMLPageAsset> versions=new ArrayList<HTMLPageAsset>(); 
-        
+
         HTMLPage working=(HTMLPage) APILocator.getVersionableAPI().findWorkingVersion(legacyident, user, respectFrontEndPermissions);
-        HTMLPageAsset cworking=copyLegacyData(working, user, respectFrontEndPermissions), clive=null;
+        HTMLPageAsset cworking = migrateLegacyData(working, user, respectFrontEndPermissions), clive=null;
         if(vInfo.getLiveInode()!=null && !vInfo.getLiveInode().equals(vInfo.getWorkingInode())) {
             HTMLPage live=(HTMLPage) APILocator.getVersionableAPI().findLiveVersion(legacyident, user, respectFrontEndPermissions);
-            clive=copyLegacyData(live, user, respectFrontEndPermissions);
+            clive = migrateLegacyData(live, user, respectFrontEndPermissions);
         }
-        
+
         List<Permission> perms=null;
         if(!APILocator.getPermissionAPI().isInheritingPermissions(legacyPage)) {
             perms = APILocator.getPermissionAPI().getPermissions(legacyPage, true, true, true);
         }
-        
+
         List<MultiTree> multiTree = MultiTreeFactory.getMultiTree(working.getIdentifier());
```

Comment:
Hey Guys, I think the purpose of the commit is to enhance existing functionality requested on https://github.com/dotCMS/core/issues/9971 (not to rewrite/refactor the code). If we do that kind of changes, we should be making them in parallel to the other code that uses the refactored code (otherwise the code will end-up messed with different patterns to do same thing). Furthermore:
1. This is prone to errors and regressions (would probably not be able to test all the functionality)
2. This moves the issue to another complexity (different than the one initially stated, I am just learning the code and taking advantage of it)
Please consider